rpc/types.go

```diff
@@ -1,66 +0,0 @@
-package rpc
-
-import (
-	"fmt"
-	"math"
-	"math/big"
-	"strings"
-
-	"github.com/ethereum/go-ethereum/common/hexutil"
-)
-
-// BlockNumber represents decoding hex string to block values
-type BlockNumber int64
-
-const (
-	// LatestBlockNumber mapping from "latest" to 0 for tm query
-	LatestBlockNumber = BlockNumber(0)
-
-	// EarliestBlockNumber mapping from "earliest" to 1 for tm query (earliest query not supported)
-	EarliestBlockNumber = BlockNumber(1)
-)
-
-func NewBlockNumber(n *big.Int) BlockNumber {
-	return BlockNumber(n.Int64())
-}
-
-// UnmarshalJSON parses the given JSON fragment into a BlockNumber. It supports:
-// - "latest", "earliest" or "pending" as string arguments
-// - the block number
-// Returned errors:
-// - an invalid block number error when the given argument isn't a known strings
-// - an out of range error when the given block number is either too little or too large
-func (bn *BlockNumber) UnmarshalJSON(data []byte) error {
```

Comment:
does this mean "latest", "pending", strings in parameters etc are no longer supported?

Reply:
are they supported on geth JSON-RPC?

Reply:
yeah they are